proxy/src/auth/password_hack.rs

```diff
@@ -0,0 +1,51 @@
+use serde::{de, Deserialize, Deserializer};
+use std::fmt;
+
+#[derive(Deserialize)]
+pub struct PasswordHackPayload {
+    pub project: String,
+
+    /// Token is base64-encoded because it may contain arbitrary byte sequences.
+    #[serde(deserialize_with = "deserialize_base64")]
+    pub token: Vec<u8>,
+}
+
+fn deserialize_base64<'a, D: Deserializer<'a>>(des: D) -> Result<Vec<u8>, D::Error> {
+    struct Visitor;
```

Comment:
Should be correct to replace the body of the function with the following code:

```rust
let base64: &str = Deserialize::deserialize(des)?;
base64::decode(base64.as_bytes()).map_err(|e| serde::de::Error::custom(e))
```

Reply:
Indeed!